openfe_gromacs/protocols/gromacs_md/md_settings.py

```diff
@@ -0,0 +1,668 @@
+        return v
+
+
+class OutputSettings(SettingsBaseModel):
+    """ "
+    Output Settings for simulations run using Gromacs
+    """
+
+    forcefield_cache: Optional[str] = "db.json"
+    """
+    Filename for caching small molecule residue templates so they can be
+    later reused.
+    """
+    mdp_file: str = "em.mdp"
+    """
+    Filename for the mdp file for running simulations in Gromacs.
+    Default 'em.mdp'
+    """
+    nstxout: int = 0
+    """
+    Number of steps that elapse between writing coordinates to the output
+    trajectory file (trr), the last coordinates are always written unless 0,
+    which means coordinates are not written into the trajectory file.
+    Default 0.
+    """
+    nstvout: int = 0
+    """
+    Number of steps that elapse between writing velocities to the output
+    trajectory file (trr), the last velocities are always written unless 0,
+    which means velocities are not written into the trajectory file.
+    Default 0.
+    """
+    nstfout: int = 0
+    """
+    Number of steps that elapse between writing forces to the output trajectory
+    file (trr), the last forces are always written, unless 0, which means
+    forces are not written into the trajectory file.
+    Default 0.
+    """
+    nstlog: int = 1000
+    """
+    Number of steps that elapse between writing energies to the log file, the
+    last energies are always written. Default 1000.
+    """
+    nstcalcenergy: int = 100
+    """
+    Number of steps that elapse between calculating the energies, 0 is never.
+    This option is only relevant with dynamics. This option affects the
+    performance in parallel simulations, because calculating energies requires
+    global communication between all processes which can become a bottleneck at
+    high parallelization. Default 100.
+    """
+    nstenergy: int = 1000
+    """"
+    Number of steps that elapse between writing energies to energy file, the
+    last energies are always written, should be a multiple of nstcalcenergy.
+    Note that the exact sums and fluctuations over all MD steps modulo
+    nstcalcenergy are stored in the energy file, so gmx energy can report
+    exact energy averages and fluctuations also when nstenergy > 1
+    """
+    nstxout_compressed: int = 0
+    """
+    Number of steps that elapse between writing position coordinates using
+    lossy compression (xtc file), 0 for not writing compressed coordinates
+    output. Default 0.
+    """
+    compressed_x_precision: int = 1000
+    """
+    Precision with which to write to the compressed trajectory file.
+    Default 1000.
+    """
+    compressed_x_grps: str = ""
+    """
+    Group(s) to write to the compressed trajectory file, by default the whole
+    system is written (if nstxout-compressed > 0).
+    """
+    energygrps: str = ""
+    """
+    Group(s) for which to write to write short-ranged non-bonded potential
+    energies to the energy file (not supported on GPUs)
+    """
```

Comment:
Because of the GPU limitation, I think it's ok here to just not have them? Unless you think there's a use for them.

Reply:
I think this could be useful for troubleshooting, e.g. to see whether interactions are excluded between certain groups or not (I've used this for SepTop). But maybe this doesn't need to be user interfacing?